src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -314,8 +314,25 @@ public int Run()
                     compilationRoots.Add(new ILCompiler.DependencyAnalysis.TrimmingDescriptorNode(linkTrimFilePath));
                 }
 
-                if (entrypointModule is { Assembly: EcmaAssembly entryAssembly })
+                // Get TypeMappingEntryAssembly from runtime knobs if specified
+                string typeMappingEntryAssembly = null;
+                foreach (var runtimeKnob in runtimeKnobs)
                 {
+                    var knobAndValue = runtimeKnob.Split('=', 2);
+                    if (knobAndValue.Length == 2 && knobAndValue[0] == "System.Runtime.InteropServices.TypeMappingEntryAssembly")
```

Comment:
Would it be cleaner to introduce a first class command line switch for this populated from the `TypeMapEntryAssembly` property (the approach in https://github.com/dotnet/runtime/issues/121138#issuecomment-3459953860)

Reply:
Added a proper switch to the tools. We'll need to add something to the SDK to set the RuntimeHostConfigurationOption when TypeMapEntryAssembly is set or require projects to set both, otherwise we only would set it when PublishAOT or PublishTrimmed are set.